loaders/consul_loader/consul_loader.go

```diff
@@ -54,9 +54,10 @@ func init() {
 type consulLoader struct {
 	cfg *cfg
 	// decoder        *consulstructure.Decoder
-	client         *api.Client
-	m              *sync.Mutex
-	lastTargets    map[string]*types.TargetConfig
+	client *api.Client
+	m      *sync.Mutex
+	// map of targets per service
+	lastTargets    map[string]map[string]*types.TargetConfig
```

Comment:
NIT: maybe it makes sense to have a struct as a key for targets per service, so that it becomes less hairy when adding items to a nested map?

```
type TargetKey struct {
    Service, Target string
}
lastTargets := make(map[TargetKey]*types.TargetConfig)
```

Reply:
In some functions I need to pass all the targets per service (for e.g loaders.Diff()), having nested maps allows to easily get that map without having to iterate over the lastTargets map and build a new one.